lmdeploy/pytorch/messages.py

```diff
@@ -46,6 +46,7 @@ class SamplingParam:
     bad_words: List[int] = field(default_factory=list)
     max_new_tokens: int = 512
     min_new_tokens: int = 0
+    response_format: Optional[str] = None
```

Comment:
What are the corresponding names in transformers and vLLM?

Reply:
No, in vllm, it is a little complex. https://github.com/vllm-project/vllm/blob/a7f65c2be93f491771aca31106f790bf381c0bad/vllm/entrypoints/openai/protocol.py#L140C5-L140C20 and https://github.com/vllm-project/vllm/blob/a7f65c2be93f491771aca31106f790bf381c0bad/vllm/entrypoints/openai/protocol.py#L215 use `response_format`, `guided_json`, `guided_choice` and so on.

While in OpenAI. It is only response_format. So, I only added one more argument `regex_format` for custom guide decoding.